### PR TITLE
Nearest selection transform does not support continuous mark types.

### PR DESCRIPTION
**Pull request by arvind**
Addresses issues raised in #3004 which identified a bug the `nearest` transform and continuous mark types (i.e., `line` and `area`). For now, this PR drops support for these mark types (which has the added effect of simplifying code paths) and the compiler issues a warning. This limitation is also noted in the documentation, with a workaround shown in an example.

### CODE REVIEW

File: site/docs/selection/nearest.md [hunk 1]
```diff
@@ -16,3 +16,7 @@ In the scatterplot below, points <select onchange="changeSpec('paintbrush_neares
 The `nearest` transform also respects any [spatial encoding projections](project.html) applied to the selection. For instance, in the example below, moving the mouse cursor back-and-forth snaps the vertical rule and label to the nearest `date` value.
 
 <div id="paintbrush_nearest" class="vl-example" data-name="stocks_nearest_index"></div>
+
+## Current Limitations
+
+* The `nearest` transform is not supported for continuous mark types (i.e., `line` and `area`). For these mark types, consider layering a discrete mark type (e.g., `point`) with a 0-value `opacity` as in the last example above.
```
Comment (domoritz) [inline site/docs/selection/nearest.md]: The example above doesn't use 0 opacity. How about "consider layering a discrete mark type (e.g., `point`) as in the last example above but use a 0-value `opacity`."

Comment (arvind) [reply site/docs/selection/nearest.md]: Not sure what you mean -- [here](https://github.com/vega/vega-lite/pull/3009/files#diff-41b1e96b395ea6629b4c649c14778f7bR28) is the 0-value `opacity` on the `point` mark.

**Comment by domoritz** [comment]
Nearest is also behaving a little funny with bars:

```
{
  "data": {"url": "data/cars.json"},
  "selection": {
    "paintbrush": {
      "type": "multi",
      "on": "mouseover",
      "nearest": true
    }
  },
  "mark": "bar",
  "encoding": {
    "x": {"field": "Horsepower", "type": "quantitative", "bin": true},
    "y": {"aggregate": "mean", "field": "Miles_per_Gallon", "type": "quantitative"},
    "color": {
      "condition": {
        "selection": "paintbrush",
        "field": "Cylinders", "type": "nominal"
      },
      "value": "grey"
    }
  }
}
```

I understand that the voronoi tesselation is using the corners but it's still a bit odd to use. 

Regardless of this PR, I wonder whether the definition of nearest could be different when the domain in ordinal. Think of a normal bar chart. If I want to use it for cross-highlighting, I could have to hover over a bar to add it to the selection. This can be hard for bars that are small. Instead, a better method is to use the background as the hit target (I did this in https://www.domoritz.de/papers/2017-TrustButVerify-CHI.pdf). Nearest could take on this semantic in this case. Just a thought, no action item.

File: site/docs/selection/nearest.md (hunk 1 above)
Comment (domoritz) [reply site/docs/selection/nearest.md]: Oh, I missed that update. 👼

Comment (domoritz) [reply site/docs/selection/nearest.md]: It's also not quite the right semantics but the best we can do. What I mean is that the closest line may not be the one that has the closest control point.

**Pull request merged**